numpy/f2py/service.py

```diff
@@ -0,0 +1,327 @@
+                        um.append(modules[mnames.index(u)])
+                    else:
+                        outmess(
+                            f'\tModule "{name}" uses nonexisting "{u}" '
+                            'which will be ignored.\n')
+            ret[name] = {}
+            _dict_append(ret[name], rules.buildmodule(module, um))
+    return ret
+
+
+def _generate_signature(postlist, sign_file: str):
+    outmess(f"Saving signatures to file {sign_file}" + "\n")
+    pyf = crackfortran.crack2fortran(postlist)
+    if sign_file in {"-", "stdout"}:
+        sys.stdout.write(pyf)
+    else:
+        with open(sign_file, "w") as f:
+            f.write(pyf)
+
+def _check_postlist(postlist, sign_file: str):
+    isusedby = {}
+    for plist in postlist:
+        if 'use' in plist:
+            for u in plist['use'].keys():
+                if u not in isusedby:
+                    isusedby[u] = []
+                isusedby[u].append(plist['name'])
+    for plist in postlist:
+        if plist['block'] == 'python module' and '__user__' in plist['name'] and plist['name'] in isusedby:
+            # if not quiet:
+            outmess(
+                f'Skipping Makefile build for module "{plist["name"]}" '
+                'which is used by {}\n'.format(
+                    ','.join(f'"{s}"' for s in isusedby[plist['name']])))
+    if(sign_file):
+        outmess(
+            'Stopping. Edit the signature file and then run f2py on the signature file: ')
+        outmess('%s %s\n' %
+                (os.path.basename(sys.argv[0]), sign_file))
+        return
+    for plist in postlist:
+        if plist['block'] != 'python module':
+            # if 'python module' not in options:
+                outmess(
+                    'Tip: If your original code is Fortran source then you must use -m option.\n')
+            # raise TypeError('All blocks must be python module blocks but got %s' % (
+            #     repr(postlist[i]['block'])))
+
+def _callcrackfortran(files: List[Path], module_name: str):
+    postlist = crackfortran.crackfortran([str(file) for file in files])
+    for mod in postlist:
+        mod["coutput"] = f"{module_name}module.c"
+        mod["f2py_wrapper_output"] = f"{module_name}-f2pywrappers.f"
+    return postlist
+
+def _set_dependencies_dist(ext_args, link_resource):
+    for dep in link_resource:
+        info = get_info(dep)
+        if not info:
+                outmess('No %s resources found in system'
+                        ' (try `f2py --help-link`)\n' % (repr(n)))
+        dict_append(ext_args, **link_resource)
+
+def get_f2py_modulename(source):
+    name = None
+    _f2py_module_name_match = re.compile(r'\s*python\s*module\s*(?P<name>[\w_]+)',
+                                        re.I).match
+    _f2py_user_module_name_match = re.compile(r'\s*python\s*module\s*(?P<name>[\w_]*?'
+                                            r'__user__[\w_]*)', re.I).match
```

Comment:
Stylistic nit, I'd prefer to have all the `regex` expressions as `CAPS_NAME` variables at the top of the file.

Reply:
Agreed, this was also a leftover from `f2py2e` making changes